src/huggingface_hub/repository.py

```diff
@@ -431,12 +436,14 @@ def clone_from(self, repo_url: str, use_auth_token: Union[bool, str, None] = Non
                 repo_url = repo_url.replace("https://", f"https://user:{token}@")
 
                 if namespace == user or namespace in valid_organisations:
+                    is_repo_private = True if repo_visibility == "private" else False
```

Comment:
Is that variable a string or a boolean, it seems pretty odd to switch from one to the other.